chain/chunks/src/chunk_cache.rs

```diff
@@ -143,23 +141,21 @@ impl EncodedChunksCache {
     }
 
     pub fn remove(&mut self, chunk_hash: &ChunkHash) -> Option<EncodedChunksCacheEntry> {
-        match self.encoded_chunks.entry(chunk_hash.clone()) {
-            Occupied(entry) => {
-                let entry = entry.remove_entry().1;
-                self.remove_incomplete_chunk(entry.header.prev_block_hash(), chunk_hash);
-                Some(entry)
-            }
-            Vacant(_) => None,
+        if let Some(entry) = self.encoded_chunks.remove(chunk_hash) {
+            self.remove_incomplete_chunk(entry.header.prev_block_hash(), chunk_hash);
+            Some(entry)
+        } else {
+            None
         }
     }
 
     // Remove the chunk from the `incomplete_chunks` map. This is an internal function.
     // Use `mark_entry_complete` instead for outside calls
     fn remove_incomplete_chunk(&mut self, prev_block_hash: CryptoHash, chunk_hash: &ChunkHash) {
-        if let Some(mut chunks) = self.incomplete_chunks.remove(&prev_block_hash) {
-            chunks.remove(chunk_hash);
-            if !chunks.is_empty() {
-                self.incomplete_chunks.insert(prev_block_hash, chunks);
+        if let Occupied(mut entry) = self.incomplete_chunks.entry(prev_block_hash) {
+            entry.get_mut().remove(chunk_hash);
+            if !entry.get().is_empty() {
+                entry.remove();
```

Comment:
I think this is buggy: `!` is not needed. Does it fail any tests?

Reply:
Oh good catch, apparently no existing test is failed. I'll add a test